prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -218,6 +219,84 @@ def canonicalize_dataset_using_columns(
             )
         return datasets.DatasetDict(dataset_dict)
 
+    def canonicalize_dataset_auto(
+        self, dataset_name: str, prompt_spec: PromptSpec, num_transform: int = 3000
+    ) -> datasets.DatasetDict:
+        """Canonicalize a dataset into a suitable text-to-text format.
+
+        Args:
+            dataset_name: The name of the dataset to canonicalize.
+            prompt_spec: A prompt whose instruction field we use to transform datasets
+            num_transform: Number to transform.
+
+        Returns:
+            A canonicalized dataset.
+        """
+        configs = datasets.get_dataset_config_names(dataset_name)
+        chosen_config = configs[0]
+
+        dataset = datasets.load_dataset(dataset_name, chosen_config).shuffle().flatten()
+
+        if "train" not in dataset:
+            raise ValueError("{dataset_name} must contain a `train` split.")
+
+        columns_mapping: dict[str, str] = {}
+        counter: dict[str, int] = {}
+        # convert flattened columns like answer.text -> answer_text
+        for col in dataset["train"].column_names:
+            new_col = col.replace(".", "_")
+            if new_col in columns_mapping.values():
+                counter[new_col] = counter.get(new_col, 0) + 1
+                new_col = f"{new_col}_{counter[new_col]}"
+            columns_mapping[col] = new_col
+        dataset = dataset.rename_columns(columns_mapping)
+
+        train_columns = dataset["train"].column_names
+        train_columns_formatted = ", ".join(train_columns)
+        dataset_description = dataset["train"].info.description
+
+        if len(dataset["train"]) == 0:
+            raise ValueError("train split is empty.")
+
+        example_rows = json.dumps(dataset["train"][0], indent=4)
+
+        self._print_divider()
+        print(f"Loaded dataset. Example rows:\n{example_rows}\n")
```

Comment:
Here, and elsewhere, can we use `logger.info` (see usage [here](https://github.com/neulab/prompt2model/blob/f2eabc1ad0e23ce774b21f48c590bedc40405ff1/prompt2model/model_trainer/callback.py#L54)) rather than `print`?